### DISCUSSION TITLE
Options: how to bail if not set?

### DISCUSSION THREAD

**Discussion by WilliamDEdwards** [pytest-dev/pytest#12723] [Q&A]
# Scenario

- When calling `request.config.getoption`
- for an option that was added using `parser.addoption`
- without `default`
- and wasn't given when invoking pytest
- pytest doesn't bail, with an "option has no default, but was not specified" error

Instead, the option defaults to `None`.

# Question

How do I work around this **natively**? Is there a sentinel I can use, like `getoption` does with `Notset`?

FWIW: I suppose I could check if any option was specified in a `pytest_configure` hook, but that's tedious for many projects.

# Example

## Snippet from `conftest.py`

```
def pytest_addoption(parser: Parser) -> None:
    parser.addoption("--test", action="store")

@pytest.fixture
def test(request: pytest.FixtureRequest) -> str:
    return request.config.getoption("--test")
```

## CLI

```
pytest
```

(No `--test` is specified.)

## Expected

pytest bails with an "option has no default, but was not specified" error.

## Result

The option defaults to `None`.

**Comment by WilliamDEdwards** [comment]
I just figured out that pytest uses argparse, which supports `required=True`.

Sorry for the noise.

**Reply by RonnyPfannschmidt** [reply]
using required=True breaks --help atm

**Comment by RonnyPfannschmidt** [comment]
the recommended way currently us to use a pytest_configure hook and raising a usage-error
